Add ARMv7 support

luusl opened the pull request:
# Pull Request

## What does this PR do?
- Pad the header on 32-bit targets so the base header size becomes 16 bytes, matching the 64-bit layout

## PR checklist
Please check if your PR fulfills the following requirements:
- [X] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [X] Have you read the contributing guidelines?
- [X] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

luusl committed:
Add padding for 32-bit pointer width in Node struct

luusl committed:
Pad the header on 32-bit targets so the base header size becomes 16 bytes, matching the 64-bit layout

Kerollmops commented on src/bbbul.rs:
```diff
@@ -58,12 +58,22 @@ struct Node {
 
 impl Node {
+    // Compute BASE_SIZE to match the actual struct layout for the current pointer width.
+    #[cfg(target_pointer_width = "64")]
```
What do you think of using the offset_of macro instead? If it's const (I haven't checked) we could keep this BASE_SIZE easier to declare.

luusl replied:
That did not work, so I split up the `Node` into `NodeHeader` and `Node` which uses `NodeHeader`, that way I got it working.

Kerollmops reviewed (changes requested):
Thank you very much for your contribution 😇

luusl committed:
Resolve code-review comment

luusl committed:
Refactor Node to use separate NodeHeader struct and add padding
on 32-bit targets to maintain consistent 16-byte header size across
architectures. Fixes BASE_SIZE calculation for unsized Node type.

Kerollmops reviewed (approved):
Thank you very much, @luusl, and sorry for the very late reply. Merging 🙏
